lib/assert.js

```diff
@@ -185,6 +185,36 @@ function isArguments(object) {
   return Object.prototype.toString.call(object) == '[object Arguments]';
 }
 
+function compareMaps(a, b, strict) {
+  // check size of the maps
+  if (a.size !== b.size)
+    return false;
+
+  // iterate over all keys and check if they exist in the other
+  for (let kvs of a) {
```

Comment:
`let` is not faster than `var`, if this code is hotcode, we would be better to use `var` instead of `let`.